test_conformance/basic/test_explicit_s2v.cpp

```diff
@@ -170,11 +141,13 @@ int test_explicit_s2v_function(cl_device_id deviceID, cl_context context, cl_com
 
     /* Now verify the results. Each value should have been duplicated four times, and we should be able to just
      do a memcpy instead of relying on the actual type of data */
-    error = clEnqueueReadBuffer( queue, streams[1], CL_TRUE, 0, destStride * count, outData, 0, NULL, NULL );
+    error =
+        clEnqueueReadBuffer(queue, streams[1], CL_TRUE, 0, destStride * count,
+                            outData.data(), 0, NULL, NULL);
     test_error( error, "Unable to read output values!" );
 
     inPtr = (unsigned char *)inputData;
-    outPtr = (unsigned char *)outData;
+    outPtr = (unsigned char *)outData.data();
```

Comment:
For half-float conversions, we need to add a NAN check in this code. i.e. the if the input is NAN the output needs to be NAN but the fraction part isn't required to be the same.